services/121-service/src/registration/repositories/registration-scoped.repository.ts

```diff
@@ -111,9 +112,22 @@ export class RegistrationScopedRepository extends RegistrationScopedBaseReposito
     return this.repository.update(criteria, partialEntity);
   }
 
-  async getRegistrationByReferenceId(referenceId: string) {
+  public async getRegistrationByReferenceId({
+    referenceId,
+    programId,
+    relations = [],
+  }: {
+    referenceId: string;
+    programId?: number;
+    relations?: string[];
+  }) {
+    const whereOptions = {
+      referenceId: Equal(referenceId),
+      ...(programId != undefined ? { programId: Equal(programId) } : {}),
```

Comment:
Does this not circumvent the security fix for including programId in the query?

Reply:
I checked in code and apparently there's situations where we don't (want?) to do this. So that's why I made optional.